blockchain/merkle.go

```diff
@@ -141,3 +140,91 @@ func BuildMerkleTreeStore(transactions []*btcutil.Tx, witness bool) []*wire.ShaH
 
 	return merkles
 }
+
+// ExtractWitnessCommitment attempts to locate, and return the witness
+// commitment for a block. The witness commitment is of the form:
+// SHA256(witness root || witness nonce).The function additionally returns a
+// boolean indicating if the witness root was located within any of the txOut's
+// in the passed transaction. The witness commitment is stored as the data push
+// for an OP_RETURN with special magic bytes to aide in location.
+func ExtractWitnessCommitment(tx *btcutil.Tx) ([]byte, bool) {
+	var witnessCommitment []byte
+
+	// The witness commitment *must* be located within one of the coinbase
+	// transaction's outputs.
+	if !IsCoinBase(tx) {
+		return witnessCommitment, false
+	}
+
+	msgTx := tx.MsgTx()
+	witFound := false
+	for i := len(msgTx.TxOut) - 1; i >= 0; i-- {
+		// The public key script that contains the witness commitment
+		// must shared a prefix with the WitnessMagicBytes, and be at
+		// least 38 bytes.
+		if bytes.HasPrefix(msgTx.TxOut[i].PkScript, WitnessMagicBytes) &&
```

Comment:
Probably doesn't matter, but I'm not really sure returning whether or not the witness commitment was found is worth it here.  It's easy enough to do `if len(witnessCommitment) == 0 { .... }`